react/src/components/tooltip/SprkTooltip.js

```diff
@@ -0,0 +1,194 @@
+      this.tooltipRef.current.setAttribute('style', 'width:' + calculatedWidth + "px");
+    }
+  }
+
+  handleWindowKeydown(e) {
+    if (e.key === 'Escape' || e.key === 'Esc' || e.keyCode === 27) {
+      this.setState({
+        isToggled: false
+      })
+    }
+  }
+
+  handleWindowClick(e) {
+    if (this.state.isToggled) {
+      if (!this.tooltipRef.current.contains(e.target)
+        && !this.triggerRef.current.contains(e.target)) {
+        this.setState({
+          isToggled: false
+        });
+      }
+    }
+  }
+
+  toggle() {
+    this.setState(prevState => ({
+      isToggled: !prevState.isToggled
+    }))
+  }
+
+  render() {
+    const {
+      children,
+      idString,
+      triggerIconType,
+      iconAdditionalClasses,
+      additionalClasses,
+      analyticsString,
+      isDefaultOpen,
+      id,
+      ...other
+    } = this.props;
+
+    return (
+      <span {...other} className='sprk-c-Tooltip__container' data-id={idString}>
+        <button
+          ref={this.triggerRef}
+          onClick={this.toggle}
+          onMouseOver={this.setPositioningClass}
+          onFocus={this.setPositioningClass}
+          className={classnames({
+            'sprk-c-Tooltip__trigger': true,
+            'sprk-c-Tooltip--toggled': this.state.isToggled,
+          })}
+          aria-expanded={this.state.isToggled ? 'true' : 'false'}
+          aria-labelledby={id}
+          data-analytics={analyticsString}
+        >
+          <SprkIcon iconName={triggerIconType} additionalClasses={iconAdditionalClasses} />
+        </button>
+        <span
+          ref={this.tooltipRef}
+          className={classnames(
+            'sprk-c-Tooltip',
+            additionalClasses,
+            {
+              'sprk-c-Tooltip--top-left': this.state.position === 'topleft',
+              'sprk-c-Tooltip--top-right': this.state.position === 'topright',
+              'sprk-c-Tooltip--bottom-left': this.state.position === 'bottomleft',
+              'sprk-c-Tooltip--bottom-right': this.state.position === 'bottomright',
+            }
+          )}
+          id={id}
+        >
+          {children}
+        </span>
+      </span>
+    );
+  }
+}
+
+SprkTooltip.defaultProps = {
+  triggerIconType: 'question-filled',
+  id: uniqueId('sprk_tooltip_'),
+};
+
+SprkTooltip.propTypes = {
+  /**
+   * Expects a space separated string
+   * of classes to be added to the
+   * tooltip element.
+   */
+  additionalClasses: PropTypes.string,
+  /**
+   * The value supplied will be assigned to the
+   * `data-analytics` attribute on the trigger element.
+   * Intended for an outside
+   * library to capture data.
+   */
+  analyticsString: PropTypes.string,
+  /**
+   * Expects a space separated string
+   * of classes to be added to the
+   * svg icon.
+   */
+  iconAdditionalClasses: PropTypes.string,
+  /**
+   * Optional: the unique ID to use for the tooltip element.
+   * If an ID is not provided, a unique ID will be created
+   * automatically.
```

Comment:
I don't think we need to let them know that it's optional with a default. They'll be able to see it in the prop table. Can probably just let them know what part of the component it will be rendered on.